R/nfunctions.R

```diff
@@ -188,55 +189,3 @@ ntype.tokens <- function(x, ...) {
         x <- tokens(x, ...)
     vapply(unclass(x), function(y) length(unique(y[y > 0])), integer(1))
 }
-
-#' Count the number of sentences
-#'
-#' Return the count of sentences in a corpus or character object.
-#' @param x a character or [corpus] whose sentences will be counted
-#' @note `nsentence()` relies on the boundaries definitions in the \pkg{stringi}
-#'   package (see [stri_opts_brkiter][stringi::stri_opts_brkiter]).  It does not
-#'   count sentences correctly if the text has been transformed to lower case,
-#'   and for this reason `nsentence()` will issue a warning if it detects all
-#'   lower-cased text.
-#' @return count(s) of the total sentences per text
-#' @examples
-#' # simple example
-#' txt <- c(text1 = "This is a sentence: second part of first sentence.",
-#'          text2 = "A word. Repeated repeated.",
-#'          text3 = "Mr. Jones has a PhD from the LSE.  Second sentence.")
-#' nsentence(txt)
-#' @export
-nsentence <- function(x) {
-    UseMethod("nsentence")
-}
-
-#' @export
-nsentence.default <- function(x) {
-    check_class(class(x), "nsentence")
-}
-
-#' @export
-#' @importFrom stringi stri_detect_charclass
-nsentence.character <- function(x) {
-    upcase <- try(any(stri_detect_charclass(x, "[A-Z]")), silent = TRUE)
-    if (!is.logical(upcase)) {
-        # warning("Input text contains non-UTF-8 characters.")
-    } else if (!upcase)
-        warning("nsentence() does not correctly count sentences in all lower-cased text")
-    ntoken(tokens(x, what = "sentence"))
```

Comment:
`nsentence.character()` and `nsentence.corpus()` are good examples of the behaviour we should deprecate or remove asap, since they have to tokenise non-tokenised objects on the fly. (Same as with `summary.corpus()`.) We should be only returning summaries on tokenised objects that are already tokenised by the user, on principle. I think we agree 100% on this. The question is just how to phase it out without breaking dependencies or alienating users too badly.

Reply:
I want to remove the function as soon as possible, but fixed the error.